### DISCUSSION TITLE
How to declare a fulltext index on multiple properties?

### DISCUSSION THREAD

**Discussion by Midnighter** [neo4j-contrib/neomodel#745] [Q&A] [answered]
Even after reading through the documentation very carefully and looking at https://github.com/neo4j-contrib/neomodel/issues/67, I still don't understand how to define a fulltext index on one or more properties. When I tried to set `index=True` on a string property, a range index was created 😕. Say, for example, I have a class defined as below. How do I create one fulltext index over both properties?

```py
from neomodel import ArrayProperty, StringProperty, StructuredNode


class Entity(StructuredNode):
    __index__ = "entity_fulltex_index"

    name = StringProperty(index=True)
    synonyms = ArrayProperty(StringProperty(), index=True)
```

Thank you for your help.  I would definitely appreciate a dedicated section in the documentation on indeces.

**Comment by mariusconjeaud** [comment]
Sorry, I didn't check this section for some time now since it was inactive. And I also didn't get any notifications, so I completely missed your question !

Full-text indexes creation is not supported ; but it is something we can add. That, and Vector indexes. Stay tuned, I'm thinking I could include it in release 5.3.2 or 5.3.3 !

**Comment by mariusconjeaud** [comment, marked as answer]
If you check the last PR into rc/5.3.2, I added fulltext index support, but only on single-property for now.
I created a specific section in the documentation for indices and constraints (called "Schema management").

Do you have suggestions about how you would like to declare it on the StructuredNode side for multi-properties ?

**Reply by Midnighter** [reply]
Thank you for the expanded documentation. That is most helpful.

Regarding multi-properties, perhaps a look at SQLAlchemy is useful. I could imagine something similar to https://docs.sqlalchemy.org/en/20/orm/declarative_tables.html#declarative-table-configuration.

```py
class City(StructuredNode):
    name = StringProperty(index=True)
    country = StringProperty(index=True)
    description = StringProperty()

    __args__ = (
        FulltextIndex(["name", "description"], analyzer="english"),
        UniqueConstraint(["name", "country"]),
    )
```

**Reply by mariusconjeaud** [reply]
OK I see ! This is worth exploring indeed ! It could an object like __schema__, containing everything schema (maybe except indexes)
